xml/cl.xml

```diff
@@ -719,6 +720,8 @@ server's OpenCL/api-docs repository.
         <enum value="((cl_device_partition_property_ext)0 - 1)"    name="CL_PARTITION_BY_NAMES_LIST_END_EXT"/>
         <enum value="-1"            name="CL_PARTITION_BY_NAMES_LIST_END_INTEL"/>
         <enum value="0"             name="CL_MEM_DEVICE_HANDLE_LIST_END_KHR"/>
+        <enum value="1"             name="CL_MEM_DEVICE_SHARED_ADDRESS_EXT"/>
+        <enum value="2"             name="CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT"/>
```

Comment:
These 2 enums are now defined in 2 places.

Reply:
Good spot, I forgot the flags. Removed.

Reply:
They aren't supposed to be flags? But below, they are added to `cl_mem_flags`:
```
            <require comment="cl_mem_flags">
                <enum name="CL_MEM_DEVICE_SHARED_ADDRESS_EXT"/>
                <enum name="CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT"/>
            </require>
```
From the definition in spec, it looks like they should be in `cl_mem_properties` instead. But then it's not obvious to me which type these properties have.

Meanwhile, in "Version history", they are called "allocation flags". If so, they are supposed to have their own enum type, and placing them in the `MiscNumbers` block is wrong.

Reply:
Thanks. No, I converted them from flags to properties after the good @kpet suggestion. I fixed the wrong type, it's now cl_mem_properties. I'm a bit confused where the cl_mem_properties enum values should be defined, so I added it where the existing one was. The type of these properties is boolean, I believe it's described in the spec text.

Reply:
Ah, ok; you can add them to a separate `<enums>` block, but it's not as important if these are not flags.

Still, if these are properties, what are the types of values they describe? I don't see this described in the spec you've added with this PR.

Personally, I find it most clear when new enums from the same type are added to a table in the spec, with enum, value type, and description columns. Like in [table 18 in core spec](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#external-buffer-memory-properties-table).

Reply:
Also, maybe it's better to allocate new unique enum values instead of using `1` and `2`, which would intersect with many things. Currently, if another extension tries to add its own enums to `cl_mem_properties` and its authors wouldn't notice that your extension does the same, it might also use values `1` and `2` for its enums, and that would make their extension arbitrarily incompatible with this one, That's why most `<enum>` tags are sorted by their value, so new enums can be added either at the end or in the gaps (`<unused>` tags, though most are reserved by specific vendors).

Reply:
> Ah, ok; you can add them to a separate `<enums>` block, but it's not as important if these are not flags.
> 
> Still, if these are properties, what are the types of values they describe? I don't see this described in the spec you've added with this PR.
> 
> Personally, I find it most clear when new enums from the same enum type (`cl_mem_properties` in our case) are added to a table in the spec, with enum, value type, and description columns. Here is how it looks in [table 18 of core spec](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#external-buffer-memory-properties-table).

Hm. I added it [right there](https://github.com/KhronosGroup/OpenCL-Docs/pull/1159/files#diff-6916ed17983ed1950fed3520537beb85d6a0153574436dde334f4911015a030aR603)? I added the enum below the existing (only) mem property that was already there (CL_MEM_DEVICE_HANDLE_LIST_KHR). I'm not sure what you mean with your improvement. Do you mean making all enums globally unique, regarless of their use/"type"?

Reply:
> Hm. I added it [right there](https://github.com/KhronosGroup/OpenCL-Docs/pull/1159/files#diff-6916ed17983ed1950fed3520537beb85d6a0153574436dde334f4911015a030aR603)?

I see, rich diff is somewhat broken for that file - but I can now see it in source text.

> Do you mean making all enums globally unique, regarless of their use/"type"?

Yes, most OpenCL enums already have globally unique values, and their names are also not re-used in different contexts.

The only enums that aren't globally unique are special values like `0` and `-1`, as well as all the flags.